src/jit/morph.cpp

```diff
@@ -7722,7 +7722,7 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
     {       
         GenTreePtr value = gtArgEntryByArgNum(call, 2)->node;
 
-        if (value->OperGet() == GT_CNS_INT && value->AsIntConCommon()->IconValue() == 0)
+        if (value->IsZero())
```

Comment:
It is foolish to have abstractions that are unused, or rarely used. In this particular context, we are looking at a particular helper call, and we know the arg type is int. We don't need to check.
I have selected the abstractions that are useful.
Please note that you may not like the name, but what I have implemented is IsIntegralZero() - and there is no need to add extra qualification when we already know the type.